ERCS/erc-7683.md

```diff
@@ -24,26 +24,47 @@ By implementing a standard, cross-chain intents systems can interoperate and sha
 
 The key words “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”, “SHOULD NOT”, “RECOMMENDED”, “MAY”, and “OPTIONAL” in this document are to be interpreted as described in RFC 2119.
 
-### CrossChainOrder struct
+### Glossary of Terms
 
-A compliant cross-chain order type MUST be ABI decodable into the `CrossChainOrder` type.
+- **Destination Chain**: the chain where the intent is executed and the user receives funds. Note: intents can involve multiple destination chains.
+- **Filler**: a participant who fulfils a user intent on the destination chain(s) and receives payment as a reward.
+- **Leg**: a portion of the user intent that can be executed independently from others. All legs must be executed for an intent to be considered fulfilled.
+- **Origin chain**: the chain where the user sends funds.
+- **Settlement System**: a system that custodies user deposits, verifies fills, and pays fillers for the purpose of faciliating intents.
+- **Settler**: a contract that implements part of the settlement system on a particular chain.
+- **User**: for the purposes of this document, the user is the end-user who is sending the ERC-7683 order.
+
+### Order structs
+
+A compliant cross-chain order type MUST be ABI decodable into either `GaslessCrossChainOrder` or `OnchainCrossChainOrder` type.
 
 ```solidity
-/// @title CrossChainOrder type
-/// @notice Standard order struct to be signed by swappers, disseminated to fillers, and submitted to settlement contracts
-struct CrossChainOrder {
+/// @title GaslessCrossChainOrder CrossChainOrder type
+/// @notice Standard order struct to be signed by users, disseminated to fillers, and submitted to origin settler contracts
+struct GaslessCrossChainOrder {
```

Comment:
As mentioned in the open question to the PR, I think it could be really helpful if we add a bytes32 type field here. The reason for this is - 
1. It would allow offchain entities to go through through a large volume of 7683 orders by simply filtering the order types they support through this type field. In a world where there are millions of intents being generated every minute, decoding the order data of each of these to understand whether you want to fill this order or not becomes unreasonable.
2. It would allow fillers to optimize resolving orders, by recreating the resolution logic of supported settlement systems offchain. Saving them an API call, and increasing speed for the end user. 
3. This would enable the creation of an onchain registry of orderData types, which we could standardize in the future as a replacement to the subtypes repo.

Reply:
Yeah, I agree with you. This is feedback we've gotten pretty frequently. The biggest downside is just bloating the order struct.

We will also need to standardize how these are generated (at least) and potentially have a registry to ensure uniqueness. My first thought is just to use a similar mechanism to EVM function and event identifiers, which are just hashes of a signature. And then we just don't worry about collisions.

Reply:
I added an orderDataType field to the standard. Right now this is just an EIP 712 typehash. However, I would ideally like this to also encapsulate the expected type for the filler data, since I think that would make things easier on fillers.

Any ideas?

Reply:
Yeah, 712 typehash is a good idea. It might be that a particular intent could be filled with multiple filler data types? 
If the user wants to commit to a particular filler data type, then they could also add this to their order data struct?